### DISCUSSION TITLE
What's the logic for when and how `tariffInformation.season` is populated?

### DISCUSSION THREAD

**Discussion by GrahamLea** [amberelectric/public-api#102] [Q&A] [answered]
In the API docs, the `season` property of the `tariffInformation` type says:
> Only available if the site in on a time of use tariff

and possible values are: `summer, autumn, winter, spring, nonSummer, holiday, weekend, weekendHoliday, weekday`

I believe I'm on a ToU tariff (Ausgrid's EA025), and I'm finding in my data the property is sometimes there, sometimes not.

What I'm seeing is:

- On a Tuesday in Summer, I see `"season": "summer"` on Usages that have a `period` of `peak` or `shoulder`, but no value for `season` for other values of `period`.
- On a Sunday in Summer, I get no `season` on any Usages. 
- On a Tuesday in Winter, I see `"season": "nonSummer"` on Usages that have a `period` of `peak` or `shoulder`, but no value for `season` for other values of `period`.
- On a Friday in Spring, I get no `season` on any Usages. 

So, that's not matching what I was expecting based on the docs.
What's the logic that determines what goes into the `season` property and whether anything goes in at all?

I was hoping I might be able to leverage this property to determine what days demand pricing would be calculated from according to my DNSP's definitions (below). But if I can't reliably tell Winter from Non-Summer via the API, I think I'll have to go with entering the date ranges myself? (Not really a question except for "Am I wrong?")

As for determining whether it's a "working weekday" or not, I think I can surmise that based on whether there's any `shoulder` or `peak` periods during a day? Would that work for all DNSPs? `offPeak` only periods for the whole day = weekend or holiday, otherwise it's working weekday?

<img width="591" alt="image" src="https://user-images.githubusercontent.com/754403/155322795-f47d4b0d-c2df-4f8c-a3b6-9ee4b7df7e2c.png">

**Comment by madpilot** [comment, marked as answer]
I hope you are enjoying this rabbit hole.

You have just discovered a slight nuance with the season modelling - it only includes seasons relevant to the network. (The docs include _all_ possible values for all tariffs, but you will only ever see a subset of them depending on your network).

>On a Tuesday in Summer, I see "season": "summer" on Usages that have a period of peak or shoulder, **but not value for season for other values of period**.
> On a Tuesday in Winter, I see "season": "nonSummer" on Usages that have a period of peak or shoulder, **but not value for season for other values of period.**

That seems a little weird - there is definitely a summer off-peak definition. I'll have a look.

> On a Sunday in Summer, I get no season on any Usages.

eg: Seasons only apply during the week, so there is no season for weekends or public holidays - they are the same all year round. 

> On a Friday in Spring, I get no season on any Usages.

Ausgrid has no concept of "spring", it's just "summer" and "winter" (They call it high and low season) - note: It looks like we've mislabeled winter "nonSummer". I'll double check. Regardless, during spring and autumn, the fallback prices kick in, so there is no season in the feed.

Now. The fun bit. The demand window time periods are not the same as the time of use time periods. In time of use land, High season is Nov - Mar, and Low Season is Jun - Aug: https://www.ausgrid.com.au/Your-energy-use/Meters/Time-of-use-pricing

So, no you can't really derive the demand windows from the TOU windows. Sorry!

**Reply by GrahamLea** [reply]
> I hope you are enjoying this rabbit hole.

You know I'm lovin' it! 😁 (If all my questions get too much, feel free to tell me so. But I'm hoping you're getting some value out of me picking the API data apart.)

I'm not surprised that the season values are some what of a pass-through from the network, seeing as the enums aren't even mutually-exclusive. The docs description just gave me a little false hope that maybe I'd get a little more value out of them than is there.

FWIW, it appears Ausgrid's High Season demand periods (above) correspond to their peak ToU periods (your link), though I'm not surprised if that isn't universal. However, in Low Season there's still a demand period, but no peak period, so you're right that I can't use one to determine the other. 😕

It wouldn't be too hard to just do it with date and time ranges and weekend detection, but the public holidays part makes it really annoying. I don't suppose you know of a reliable way to ask the API data "Was this a weekend _or public holiday_?"
